src/AbstractAlgebra.jl

```diff
@@ -592,7 +592,6 @@ import .Generic: cycles
 import .Generic: defining_polynomial
 import .Generic: degrees
 import .Generic: dense_matrix_type
-import .Generic: dense_poly_type
```

Comment:
Renaming `dense_poly_type` to `poly_type` is a breaking change. I would recommend against doing this in this PR, which is about `polynomial_ring`. It complicates the review, may slow down the merging etc.

We can then seperately discuss it. If we agree to do it, I would still recommend that we add `@alias dense_poly_type poly_type` at least for a time, as that makes migration much easier. Just like we did with the other renamings.